lua/rocks-config/internal.lua

```diff
@@ -208,14 +208,119 @@ function rocks_config.configure(rock, config)
+    if type(bundle.config) ~= "nil" and type(bundle.config) ~= "string" then
+        vim.schedule(function()
+            vim.notify(
+                string.format(
+                    "[rocks-config.nvim]: Bundle '%s' has invalid `config` variable. Expected string pointing to a valid path, got %s instead...",
+                    bundle_name,
+                    type(bundle.config)
+                ),
+                vim.log.levels.ERROR
+            )
+        end)
+        bundle.config = nil
+    end
+    local mod_name = bundle.config ~= nil and bundle.config
+        or table.concat({ config.config.plugins_dir, bundle_name }, ".")
+
+    local result, err = try_load_config(mod_name)
+    if result then
+        for _, plugin in ipairs(bundle.items) do
+            _configured_rocks[plugin] = true
+        end
+    elseif result == nil and type(err) == "string" then
+        vim.notify(
+            string.format(
+                [[
+[rocks-config.nvim]: Bundle '%s' failed to load ('checkhealth rocks-config' for details).
+Falling back to loading plugins from the bundle individually...
+]],
+                bundle_name
+            ),
+            vim.log.levels.WARN
+        )
+        table.insert(rocks_config.failed_to_load, { bundle_name, vim.inspect(bundle.items), err })
+    else
+        vim.notify(
+            string.format(
+                [[
+[rocks-config.nvim]: Bundle '%s' has no specified configuration file.
+Falling back to loading plugins from the bundle individually...
+]],
+                bundle_name
+            ),
+            vim.log.levels.WARN
+        )
+    end
+end
+
+---@param bundle_name string The name of the bundle
+function rocks_config.load_bundle(bundle_name)
+    local config = get_config()
+    local bundle = vim.iter(config.bundles or {}):fold(nil, function(acc, name, bundle)
+        if name == bundle_name then
+            acc = bundle
+        end
+        return acc
+    end)
+    if not bundle then
+        vim.schedule(function()
+            vim.notify(string.format("[rocks-config.nvim]: Bundle '%s' not found.", bundle_name), vim.log.levels.ERROR)
+        end)
+        return
+    end
+
+    ---@param rock_name string
+    ---@return RockSpec | nil
+    local function get_rock(rock_name)
+        return get_rock_from_config(config, rock_name)
+    end
+    ---@param item string
+    local nonexistent_bundle_item = vim.iter(bundle.items):find(function(item)
+        return get_rock(item) == nil
+    end)
+    if nonexistent_bundle_item then
+        vim.schedule(function()
+            vim.notify(
+                string.format(
+                    "[rocks-config.nvim]: Bundle '%s' has invalid plugin '%s'. Did you make a typo, or is the plugin not installed?",
+                    bundle_name,
+                    nonexistent_bundle_item
+                ),
+                vim.log.levels.ERROR
+            )
+        end)
+        return
+    end
```

Comment:
question: Should we skip loading the bundle if **all** of the items are `opt` or if **any** of them are?

Reply:
I think **any** makes the most sense. rocks-config could provide an API to get the remaining plugins in a bundle. Then modules like rocks-dev.nvim could use that to add the plugins to the rtp before sourcing the bundle.

Reply:
Done.